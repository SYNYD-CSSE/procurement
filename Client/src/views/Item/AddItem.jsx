import React, { Component } from "react";
import axios from "axios";
import {
  Row,
  Col,
  Button,
  Card,
  CardHeader,
  CardFooter,
  CardBody,
  FormGroup,
  Label,
  Input
} from "reactstrap";
import alertify from "alertifyjs";
class AddItem extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: "",
      quantity: "1",
      unit: ""
    };

    this.inputHandler = this.inputHandler.bind(this);
    this.submitHandler = this.submitHandler.bind(this);
  }

  inputHandler(e) {
    this.setState({ [e.target.name]: e.target.value });
  }
  submitHandler(e) {
    e.preventDefault();
    const { quantity, name, unit } = this.state;

    axios
      .post("http://localhost:5000/items", {
        name,
        quantity,
        unit
      })
      .then(result => {
        console.log(result);
        alertify.notify("New Item Added!", "success", 5, function() {
          console.log("dismissed");
        });
      });
    this.props.history.push("/itemList");
  }

  render() {
    return (
      <div className="container">
        <Row>
          <Col sm="10" xs="6">
            <form onSubmit={this.submitHandler}>
              <Card>
                <CardHeader>
                  <strong>ADD ITEM</strong>
                  <small> Form</small>
                </CardHeader>

                <CardBody>
                  <div className="row">
                    <div className="col-md-6">
                      <FormGroup row>
                        <Col xs="8">
                          <Label htmlFor="name">Item Name</Label>
                          <Input
                            type="text"
                            id="name"
                            name="name"
                            placeholder="Item Name"
                            onChange={this.inputHandler}
                            value={this.state.name}
                            required
                          />
                        </Col>
                      </FormGroup>

                      <FormGroup row>
                        <Col xs="8">
                          <Label htmlFor="quantity">Quantity</Label>
                          <Input
                            type="number"
                            id="quantity"
                            name="quantity"
                            placeholder="Enter Quantity"
                            onChange={this.inputHandler}
                            value={this.state.quantity}
                            required
                          />
                        </Col>
                      </FormGroup>
                      <FormGroup row>
                        <Col xs="8">
                          <Label htmlFor="unit">Unit</Label>
                          <Input
                            type="text"
                            id="unit"
                            name="unit"
                            placeholder="Enter Unit"
                            onChange={this.inputHandler}
                            value={this.state.unit}
                            required
                          />
                        </Col>
                      </FormGroup>
                    </div>
                  </div>
                </CardBody>
                <CardFooter>
                  <div className="float-right">
                    <Button type="submit" size="sm" color="primary">
                      <i className="fa fa-dot-circle-o" /> Submit
                    </Button>
                    <Button
                      type="reset"
                      size="sm"
                      color="danger"
                      onClick={this.resetHandler}
                    >
                      <i className="fa fa-ban" /> Reset
                    </Button>
                  </div>
                </CardFooter>
              </Card>
            </form>
          </Col>
        </Row>
      </div>
    );
  }
}

export default AddItem;
