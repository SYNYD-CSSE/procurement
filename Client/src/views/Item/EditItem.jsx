import React, { Component } from "react";
import axios from "axios";
import {
  Row,
  Col,
  Button,
  Card,
  CardHeader,
  CardFooter,
  CardBody,
  FormGroup,
  Label,
  Input
} from "reactstrap";
import alertify from "alertifyjs";
class EditItem extends Component {
  constructor(props) {
    super(props);
    this.state = {
      // name: "",
      // quantity: "",
      // unit: ""
      items: {}
    };

    this.inputHandler = this.inputHandler.bind(this);
    this.submitHandler = this.submitHandler.bind(this);
  }

  componentDidMount() {
    axios
      .get("http://localhost:5000/items/" + this.props.match.params.itemId)
      .then(response => {
        this.setState({ items: response.data });
      })
      .catch(function(error) {
        console.log(error);
      });
  }

  inputHandler(e) {
    // this.setState({ [e.target.name]: e.target.value });
    const state = this.state.items;
    state[e.target.name] = e.target.value;
    this.setState({ items: state });
  }
  submitHandler(e) {
    e.preventDefault();
    const { name, unit } = this.state.items;

    axios
      .put("http://localhost:5000/items/" + this.props.match.params.itemId, {
        name,
        quantity,
        unit
      })
      .then(result => {
        console.log(result);
        alertify.notify("Item Updated!", "success", 5, function() {
          console.log("dismissed");
        });
      });
    this.props.history.push("/itemList");
  }

  render() {
    return (
      <div className="container">
        <Row>
          <Col sm="10" xs="6">
            <form onSubmit={this.submitHandler}>
              <Card>
                <CardHeader>
                  <strong>EDIT ITEM</strong>
                  <small> Form</small>
                </CardHeader>

                <CardBody>
                  <div className="row">
                    <div className="col-md-6">
                      <FormGroup row>
                        <Col xs="8">
                          <Label htmlFor="name">Item Name</Label>
                          <Input
                            type="text"
                            id="name"
                            name="name"
                            placeholder="Item Name"
                            onChange={this.inputHandler}
                            value={this.state.items.name}
                            required
                          />
                        </Col>
                      </FormGroup>

                      <FormGroup row>
                        <Col xs="8">
                          <Label htmlFor="unit">Unit</Label>
                          <Input
                            type="text"
                            id="unit"
                            name="unit"
                            placeholder="Enter Unit"
                            onChange={this.inputHandler}
                            value={this.state.items.unit}
                            required
                          />
                        </Col>
                      </FormGroup>
                    </div>
                  </div>
                </CardBody>
                <CardFooter>
                  <div className="float-right">
                    <Button type="submit" size="sm" color="primary">
                      <i className="fa fa-dot-circle-o" /> Update
                    </Button>
                  </div>
                </CardFooter>
              </Card>
            </form>
          </Col>
        </Row>
      </div>
    );
  }
}

export default EditItem;
