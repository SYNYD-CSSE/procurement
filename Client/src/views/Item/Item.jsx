import React, { Component } from "react";
import { Link } from "react-router-dom";
import axios from "axios";
import alertify from "alertifyjs";

const user = JSON.parse(localStorage.getItem('user'));
const token = JSON.parse(localStorage.getItem('token'));
axios.defaults.headers.common['Authorization'] = token;

class Item extends Component {
  constructor(props) {
    super(props);
    console.log(props);
    this.deleteItem = this.deleteItem.bind(this);
  }

  deleteItem(event) {
    event.preventDefault();
    axios
      .delete("http://localhost:5000/items/" + this.props.itemId)
      .then(res => {
        alertify.notify("Item Deleted!", "success", 5, function() {
          console.log("dismissed");
        });
      })
      .catch(err => console.log(err));
  }

  //   sendToLocal() {
  //     localStorage.clear();

  //     const item = {
  //       pid: this.props.pId,
  //       bht: this.props.bht,
  //       name: this.props.name,
  //       wardNo: this.props.wardNo,
  //       bedNo: this.props.bedNo
  //     };

  //     localStorage.setItem("patientDetails", JSON.stringify(patient));
  //   }

  render() {
    return (
      <tr>
        <td>{this.props.itemId}</td>
        <td>{this.props.name}</td>
        <td>{this.props.unit}</td>
        <td>
          {/* <Link to="/bht/profile">
            <button
              onClick={this.sendToLocal.bind(this)}
              type="button"
              className="btn btn-success btn-sm"
            >
              <i className="fa fa-search fa-sm" /> View BHT
            </button>
          </Link> */}

          <Link
            to={"/editItem/" + this.props.itemId}
            className="btn btn-primary"
          >
            Edit
          </Link>
        </td>
        <td>
          <form onSubmit={this.deleteItem}>
            <input type="submit" value="Delete" className="btn btn-danger" />
          </form>
        </td>
      </tr>
    );
  }
}

export default Item;
