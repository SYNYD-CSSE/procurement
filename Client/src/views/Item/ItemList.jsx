import React, { Component } from "react";
import axios from "axios";
import Item from "./Item";
import SearchBar from "./Search";
import { Table } from "reactstrap";

class ItemList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      itemList: []
    };
  }

  componentWillMount() {
    axios
      .get("http://localhost:5000/items")
      .then(res => {
        this.setState({ itemList: res.data });
        console.log(this.state.itemList);
      })
      .catch(err => {
        console.log(err);
      });
  }



  render() {
    var array = this.state.itemList;
    var tabRow = "";

    if (array) {
      tabRow = array.map((item, i) => {
        console.log(item);

        return (
          <Item
            key={i}
            itemId={item.itemId}
            name={item.name}
            unit={item.unit}
            quantity={item.quantity}
          />
        );
      });
    } else {
      tabRow = " ";

      return tabRow;
    }
    return (
      <div className="container">
        <h4>ITEMS</h4>
        <br />
        <Table hover bordered striped responsive size="sm">
          <thead className="thead-dark">
            <tr>
              <th>Item Code</th>
              <th>Item Name</th>
              <th>Unit</th>
              <th>Quantity</th>
              <th />
              <th />
            </tr>
          </thead>
          <tbody>{tabRow}</tbody>
        </Table>
      </div>
    );
  }
}

export default ItemList;
