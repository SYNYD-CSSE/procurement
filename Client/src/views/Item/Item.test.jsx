import React from "react";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import { spy } from 'sinon';
// import sinon from "sinon";
import AddItem from "./AddItem";
import Item from "./Item";
import ItemList from "./ItemList";
import EditItem from "./EditItem"

describe("AddItem", () => {
  it("should render <AddItem> component", () => {
    const wrapper = shallow(<AddItem />);
  });

  it('should have a `<form>` element', () => {
    expect(
      wrapper.find('form').length
    ).toBe(1);
  }); 
});

describe("Item", () => {
  it("should render <Item> component", () => {
    const wrapper = shallow(<Item />);
  });

  it("should render correctly", () => {
    const output = shallow(<Item title="mockTitle" url="mockUrl" />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});

describe("<ItemList/>", () => {
  it("should  find all <Item/> component", () => {
    const wrapper = shallow(<ItemList />);
    expect(wrapper.find(Item));
  });



  it('should handle state changes', () => {
    const output = shallow(
      <ItemList title="mockTitle" url="mockUrl" />
    );
    
    expect(output.state().itemList);
  });
});
