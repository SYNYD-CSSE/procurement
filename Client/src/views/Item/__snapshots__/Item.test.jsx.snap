// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should render correctly 1`] = `
<tr>
  <td />
  <td />
  <td />
  <td>
    <Link
      className="btn btn-primary"
      replace={false}
      to="/editItem/undefined"
    >
      Edit
    </Link>
  </td>
  <td>
    <form
      onSubmit={[Function]}
    >
      <input
        className="btn btn-danger"
        type="submit"
        value="Delete"
      />
    </form>
  </td>
</tr>
`;
